internal/providers/pluginfw/products/catalog/resource_function.go

```diff
@@ -0,0 +1,204 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	pluginfwcontext "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/context"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/converters"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/databricks/terraform-provider-databricks/internal/service/catalog_tf"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+)
+
+const resourceName = "function"
+
+var _ resource.ResourceWithConfigure = &FunctionResource{}
+
+func ResourceFunction() resource.Resource {
+	return &FunctionResource{}
+}
+
+type FunctionResource struct {
+	Client *common.DatabricksClient
+}
+
+func (r *FunctionResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = pluginfwcommon.GetDatabricksProductionName(resourceName)
+}
+
+func (r *FunctionResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	attrs, blocks := tfschema.ResourceStructToSchemaMap(catalog_tf.FunctionInfo{}, func(c tfschema.CustomizableSchema) tfschema.CustomizableSchema {
+		c.ToNestedBlockObject()
+
+		c.SetRequired("name")
+		c.SetRequired("catalog_name")
+		c.SetRequired("schema_name")
+		c.SetRequired("input_params")
+		c.SetRequired("data_type")
+		c.SetRequired("full_data_type")
+		c.SetRequired("routine_defintion")
+		c.SetRequired("is_deterministic")
+		c.SetRequired("is_null_call")
+		c.SetRequired("specific_name")
+		c.SetRequired("routine_body")
+		c.SetRequired("security_type")
+		c.SetRequired("sql_data_access")
+		c.SetRequired("parameter_style")
+
+		c.SetReadOnly("full_name")
+		c.SetReadOnly("created_at")
+		c.SetReadOnly("created_by")
+		c.SetReadOnly("updated_at")
+		c.SetReadOnly("updated_by")
+
+		return c
```

Comment:
I looked at UpdateFunction, and it seems like the only thing you can update about a function is its owner. Everything else cannot be updated. Can we mark all other fields with `SetForceNew()`?

Reply:
`SetForceNew()` doesn't seem to be a method of `tfschema.CustomizableSchema`. Am I looking at the wrong place? Any guidance here appreciate it.